src/main/scala/com/fulcrumgenomics/commons/io/CommandLineTool.scala

```
@@ -0,0 +1,216 @@
+
+  /** Executes a script stored at a Path if the this executable is available.
+    *
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param script Path of the script to be run
+    * @param args a variable list of arguments to pass to the script
+    */
+  def execIfAvailable(script: Path, args: String*): Unit = execIfAvailable(script.toString, args: _*)
+
+  /** Executes a script from the classpath, raise an exception otherwise.
+    *
+    * @throws ToolException when the exit code from the called process is not zero.
+    * @throws Exception when we are unable to execute to this script on the classpath with the given arguments.
+    * @param scriptResource the name of the script resource on the classpath
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(scriptResource: String, args: String*): Unit = exec(writeResourceToTempFile(scriptResource), args: _*)
+
+  /** Executes a script from the filesystem Path.
+    *
+    * @throws ToolException when the exit code from the called process is not zero
+    * @throws Exception when we are unable to execute the script with the given arguments
+    * @param script Path to the script to be executed
+    * @param args a variable list of arguments to pass to the script
+    */
+  def exec(script: Path, args: String*): Unit = try {
+    val command = Executable +: script.toAbsolutePath.toString +: args
+    val process = new ProcessBuilder(command:_*).redirectErrorStream(false).start()
+    val pipe1   = Io.pipeStream(process.getErrorStream, logger.info)
+    val pipe2   = Io.pipeStream(process.getInputStream, logger.debug)
+    val retval  = process.waitFor()
+    pipe1.close()
+    pipe2.close()
+
+    if (retval != 0) throw ToolException(retval)
+  } catch {case e: Exception => throw new Exception(s"Cannot execute script from path $script with args: $args") }
```

The prior line (131) will throw a specific exception if the return value is non-zero. However, on this line we catch all exceptions and throw away our custom `ToolException`.

Exception handling could use some work here.

---

Also, this needs to be split across multiple lines